From b6ac6ab5131cc3b042fc473422e2a6d245ab96de Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Thu, 17 Sep 2026 11:42:27 +0200
Subject: [PATCH] test-camel-provider-imapx: Replace strcasestr() with
 camel_strstrcase()

strcasestr is not POSIX, use camel_strstrcase instead which does the
same. See this error on Alpine with musl libc:

	[77/1125] Building C object src/camel/tests/CMakeFiles/test-camel-provider-imapx.dir/test-camel-provider-imapx.c.o
	ninja: job failed with status 1: /usr/lib/ccache/bin/cc -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_68 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_68 -DG_LOG_DOMAIN=\"test-camel-provider-imapx\" -DLDAP_DEPRECATED -DSOUP_VERSION_MAX_ALLOWED=SOUP_VERSION_3_2 -DSOUP_VERSION_MIN_REQUIRED=SOUP_VERSION_3_2 -I/tmp/src/evolution-data-server-3.62.0/build -I/tmp/src/evolution-data-server-3.62.0/build/src -I/tmp/src/evolution-data-server-3.62.0/build/src/camel -I/tmp/src/evolution-data-server-3.62.0/src -I/tmp/src/evolution-data-server-3.62.0/src/camel -I/tmp/src/evolution-data-server-3.62.0/src/camel/tests/lib -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/nss -I/usr/include/nspr -Wnested-externs -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers -Wdeclaration-after-statement -Werror-implicit-function-declaration -Wno-missing-include-dirs -Wno-deprecated-declarations -fno-strict-aliasing -Wno-cast-function-type -Wwrite-strings -Wunused-function -Wundef -Wshadow -Wredundant-decls -Wpointer-arith -Wmissing-noreturn -Wmissing-declarations -Winit-self -Wformat-security -Wformat -Os -fstack-clash-protection -Wformat -Werror=format-security -fno-plt -I/usr/include/gnu-libiconv -fPIE -I/usr/include -I/usr/include/libmount -I/usr/include/blkid -pthread -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/nss -I/usr/include/nspr -MD -MT src/camel/tests/CMakeFiles/test-camel-provider-imapx.dir/test-camel-provider-imapx.c.o -MF src/camel/tests/CMakeFiles/test-camel-provider-imapx.dir/test-camel-provider-imapx.c.o.d -o src/camel/tests/CMakeFiles/test-camel-provider-imapx.dir/test-camel-provider-imapx.c.o -c /tmp/src/evolution-data-server-3.62.0/src/camel/tests/test-camel-provider-imapx.c
	/tmp/src/evolution-data-server-3.62.0/src/camel/tests/test-camel-provider-imapx.c: In function 'test_server_has_capability':
	/tmp/src/evolution-data-server-3.62.0/src/camel/tests/test-camel-provider-imapx.c:604:13: error: implicit declaration of function 'strcasestr'; did you mean 'strcasecmp'? [-Wimplicit-function-declaration]
	  604 |         if (strcasestr (buf, capability_name)) {
	      |             ^~~~~~~~~~
	      |             strcasecmp
	/tmp/src/evolution-data-server-3.62.0/src/camel/tests/test-camel-provider-imapx.c:604:13: warning: nested extern declaration of 'strcasestr' [-Wnested-externs]
	ninja: subcommand failed

Fixes: 4a52e0a82fa7 ("I#132 - Implement IMAP4 Multimailbox SEARCH Extension")
---
 src/camel/tests/test-camel-provider-imapx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/camel/tests/test-camel-provider-imapx.c b/src/camel/tests/test-camel-provider-imapx.c
index 8118212e2..57843fdb3 100644
--- a/src/camel/tests/test-camel-provider-imapx.c
+++ b/src/camel/tests/test-camel-provider-imapx.c
@@ -604,7 +604,7 @@ test_server_has_capability (const gchar *capability_name)
 	}
 	buf[nread] = '\0';
 
-	if (strcasestr (buf, capability_name)) {
+	if (camel_strstrcase (buf, capability_name)) {
 		found = TRUE;
 	} else {
 		cap_cmd = g_strdup ("a2 CAPABILITY\r\n");
@@ -612,7 +612,7 @@ test_server_has_capability (const gchar *capability_name)
 			nread = read (fd, buf, sizeof (buf) - 1);
 			if (nread > 0) {
 				buf[nread] = '\0';
-				if (strcasestr (buf, capability_name)) {
+				if (camel_strstrcase (buf, capability_name)) {
 					found = TRUE;
 				}
 			}
-- 
GitLab

